src/backend/api/meals.js

```diff
@@ -1,12 +1,166 @@
+const { query } = require("express");
 const express = require("express");
+const { where } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    let mealArrayToChange = await knex
```

Comment:
Is there any particular reason that you didn't use knex chainable methods (e.g. knex.select and knex.where)?

Reply:
@jslambda yes, for parameter “availableReservation” I need to do complex query with JOIN SUM and HAVING and this “having” doesn’t work in chainable methods. I tried to google that problem but did not find solution.  So I used knex.raw.